src/Symfony/Component/DependencyInjection/Loader/FileLoader.php

```diff
@@ -34,4 +39,116 @@ public function __construct(ContainerBuilder $container, FileLocatorInterface $l
 
         parent::__construct($locator);
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setCurrentDir($dir)
+    {
+        $this->currentDir = $dir;
+        parent::setCurrentDir($dir);
+    }
+
+    public function registerClasses(Definition $prototype, $psr4Glob, $directory)
```

Comment:
Imo this is cool on methods because of the incoming 7.0 type hints: it will allow us to upgrade a bigger part of the codebase.
However, if [this RFC](https://wiki.php.net/rfc/parameter-no-type-variance) is voted, it will be less useful.